test/functional/tests/example/example_test.py

```diff
@@ -19,37 +19,67 @@ def setup_module():
-        part_sizes.append(Size(10 * i + 100, Unit.MebiByte))
-    test_disk.create_partitions(part_sizes)
-    TestRun.LOGGER.info(f"DUT info: {TestRun.dut}")
-    test_disk.partitions[0].create_filesystem(Filesystem.ext3)
+    """
+        title: Example test creating partitions and filesystems.
+        description: Create 6 partitions and create filesystem on each of them.
+        pass_criteria:
+          - Partitions are created with no error.
+          - Filesystems are created successfully.
+    """
+    with TestRun.step("Prepare"):
+        test_disk = TestRun.disks['cache']
+
+    with TestRun.group("Repartition disk"):
+        with TestRun.step("Genetare partitions table"):
+            part_sizes = []
+            for i in range(1, 6):
+                part_sizes.append(Size(10 * i + 100, Unit.MebiByte))
+        with TestRun.step("Create partitions"):
+            test_disk.create_partitions(part_sizes)
+        for i in TestRun.iteration(range(0, 5)):
+            with TestRun.step(f"Create filesystem on partition {i}"):
+                test_disk.partitions[i].create_filesystem(Filesystem.ext3)
+
 
 
 def test_create_example_files():
-    TestRun.LOGGER.info("Test run")
-    file1 = File.create_file("example_file")
-    file1.write("Test file\ncontent line\ncontent")
-    content_before_change = file1.read()
-    TestRun.LOGGER.info(f"File content: {content_before_change}")
-    fs_utils.replace_in_lines(file1, 'content line', 'replaced line')
-
-    content_after_change = file1.read()
-    assert content_before_change != content_after_change
-
-    file2 = file1.copy('/tmp', force=True)
-    assert file1.md5sum() == file2.md5sum()
-
-    file2.chmod_numerical(123)
-    fs_utils.remove(file2.full_path, True)
-    dir1 = Directory("~")
-    dir_content = dir1.ls()
-    file1.chmod(fs_utils.Permissions['r'] | fs_utils.Permissions['w'], fs_utils.PermissionsUsers(7))
-    for item in dir_content:
-        TestRun.LOGGER.info(f"Item {str(item)} - {type(item).__name__}")
-    fs_utils.remove(file1.full_path, True)
+    """
+        title: Example test manipulating on filesystem.
+        description: Perform various operaations on filesystem.
+        pass_criteria:
+          - System does not crash.
+          - All operations complete successfully.
+          - Data consistency is being preserved.
+    """
+    with TestRun.step("Create file with content"):
+        file1 = File.create_file("example_file")
+        file1.write("Test file\ncontent line\ncontent")
+    with TestRun.step("Read file content"):
+        content_before_change = file1.read()
+        TestRun.LOGGER.info(f"File content: {content_before_change}")
+    with TestRun.step("Replace single line in file"):
+        fs_utils.replace_in_lines(file1, 'content line', 'replaced line')
+    with TestRun.step("Read file content and check if it changed"):
+        content_after_change = file1.read()
+        assert content_before_change != content_after_change
```

Comment:
please do not use assert but TestRun.fail()

Reply:
Done.